devlib/instrument/monsoon.py

```diff
@@ -0,0 +1,170 @@
+import devlib
+import json
+import os
+import os.path
+import psutil
+import time
+import logging
+from subprocess import Popen, STDOUT
+from threading import Timer
+import pandas as pd
+import StringIO
+from collections import namedtuple
+from subprocess import Popen, PIPE, STDOUT
+from devlib.instrument import Instrument, CONTINUOUS, MeasurementsCsv
+from devlib.exception import HostError
+from devlib.utils.misc import which
+
+class MonsoonInstrument(Instrument):
+    MONSOON_BIN = "./tools/scripts/monsoon.par"
+
+    def log_error(self, errmsg, exception=0):
+	print "ERROR: " + errmsg
+	self.logger.error(errmsg)
+	if exception == 1:
+		raise(Exception(errmsg))
+
+    def log_info(self, msg):
+	print "INFO: " + msg
+	self.logger.info(msg)
+
+    def __init__(self, target, device_entry=None):
+	super(MonsoonInstrument, self).__init__(target)
+
+	if not os.path.exists(MonsoonInstrument.MONSOON_BIN):
+		self.log_error("monsoon.par not found", exception=1)
+
+	if device_entry != None and not os.path.exists(device_entry):
+		self.log_error(str(device_entry) + " doesn't exist", exception=1)
+
+	self.device_entry = device_entry
+	self.process = None
+
+	self.log_info('Found monsoon.par')
+
+    def reset(self):
+        self.log_info('Initiailzing MonsoonInstrument')
+	# Waiting for monsoon to get ready
+	self.run_monsoon(['--status'], timeout=3)
+	self.log_info("MonsoonInstrument ready for action")
+
+    def kill_monsoon(self, p):
```

Comment:
For the Monsoon device we can probably support (at least initially) only the CONTINUOUS interface, thus you need to implement only 4 public methods: `reset`, `start`, `stop` and `get_data`.

Here is an example:
https://github.com/ARM-software/devlib/blob/master/devlib/instrument/energy_probe.py#L35